mozetl/engagement/churn/job.py

```diff
@@ -139,15 +170,30 @@ def extract(main_summary, new_profile, start_ds, period, slack, is_sampled):
     if is_sampled:
         predicates.append((F.col("sample_id") == "57"))
 
-    return (
+    extract_ms = (
         main_summary
         .where(reduce(operator.__and__, predicates))
         .select(SOURCE_COLUMNS)
-        .union(
-            clean_new_profile(new_profile)
-            .where(reduce(operator.__and__, predicates))
-            .select(SOURCE_COLUMNS)
-        )
+    )
+
+    np = clean_new_profile(new_profile)
+    extract_np = (
+        np
+        .where(reduce(operator.__and__, predicates))
+        .select(SOURCE_COLUMNS)
+    )
+
+    np_attribution = (
+        np
+        .where("attribution is not null")
+        .groupBy("client_id")
+        .agg(F.first("attribution").alias("attribution"))
```

Comment:
Can you confirm that `attribution` is stable across `new_profile` pings?

Reply:
It looks like there can be [multiple attribution codes per client](https://sql.telemetry.mozilla.org/queries/49745) in the new-profile pings. It looks like an overwhelming number of clients [appear once with a single configuration](https://sql.telemetry.mozilla.org/queries/49746/source#133842). I don't think there's a good way of getting around this without fixing attribution at the client level.

Reply:
Okay, it seems fine to me to proceed by using `first` in this case. Go ahead and publish that attribution query, it seems useful to have around.